apps/src/BriskMatch.cpp

```diff
@@ -0,0 +1,127 @@
+
+#include <opencv2/core.hpp>
+#include <opencv2/features2d.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgcodecs.hpp>
+
+#include <iostream>
+#include <vector>
+
+#define IMG_QUALITY 95
+
+int main(int argc, const char* argv[])
+{
+    // vectors for match type and brisk
+    std::vector<cv::String> typeDesc;
+    std::vector<cv::String> typeAlgoMatch;
+    // Vector for images being compared
+    std::vector<cv::String> fileName;
+
+    typeDesc.push_back("BRISK");
+    typeAlgoMatch.push_back("BruteForce");
+
+    cv::CommandLineParser parser(
+        argc, argv,
+        "{ @image1 | ../data/basketball1.jpg | }"
+        "{ @image2 | ../data/basketball2.jpg | }");
+
+    // names of images to be compared
+    fileName.push_back(parser.get<std::string>(0));
+    fileName.push_back(parser.get<std::string>(1));
+
+    // create Mat for images
+    cv::Mat img1 = cv::imread(fileName[0], cv::IMREAD_GRAYSCALE);
+    cv::Mat img2 = cv::imread(fileName[1], cv::IMREAD_GRAYSCALE);
+
+    // check that images opened correctly
+    if (img1.rows * img1.cols <= 0) {
+        std::cout << "Image " << fileName[0]
+                  << " is empty or cannot be found\n";
+        return (0);
+    }
+
+    if (img2.rows * img2.cols <= 0) {
+        std::cout << "Image " << fileName[1]
+                  << " is empty or cannot be found\n";
+        return (0);
+    }
+
+    std::cout << "Matching " << fileName[0] << " and " << fileName[1]
+              << std::endl;
+
+    // Not sure what this vector does
+    std::vector<double> desMethCmp;
+    cv::Ptr<cv::Feature2D> b;
+
+    std::vector<cv::String>::iterator itMatcher = typeAlgoMatch.end();
+
+    cv::Ptr<cv::DescriptorMatcher> descriptorMatcher;
+    // Match between img1 and img2
+    std::vector<cv::DMatch> matches;
+    // keypoint  for img1 and img2
+    std::vector<cv::KeyPoint> keyImg1, keyImg2;
+    // Descriptor for img1 and img2
+    cv::Mat descImg1, descImg2;
+
+    b = cv::BRISK::create();
+
+    try {
+        // detect keypoints and compute their descriptors for each image
+        b->detectAndCompute(img1, cv::Mat(), keyImg1, descImg1, false);
+        b->detectAndCompute(img2, cv::Mat(), keyImg2, descImg2, false);
+
+        itMatcher = typeAlgoMatch.begin();
+        descriptorMatcher = cv::DescriptorMatcher::create(*itMatcher);
+        try {
```

Comment:
delete this internal `try-catch` wrapper. I only want the previous one for right now.